Framework/Geometry/src/Instrument.cpp

```diff
@@ -1237,6 +1238,61 @@ void Instrument::setDetectorInfo(
     boost::shared_ptr<const Beamline::DetectorInfo> detectorInfo) {
   m_detectorInfo = std::move(detectorInfo);
 }
+
+/// Returns the index for a detector ID. Used for accessing DetectorInfo.
+size_t Instrument::detectorIndex(const detid_t detID) const {
+  const auto &baseInstr = m_map ? *m_instr : *this;
+  const auto it = find(baseInstr.m_detectorCache, detID);
+  return std::distance(baseInstr.m_detectorCache.cbegin(), it);
+}
+
+/// Returns a legacy ParameterMap, containing information that is now stored in
+/// DetectorInfo (masking, positions, rotations).
+boost::shared_ptr<ParameterMap> Instrument::makeLegacyParameterMap() const {
+  auto pmap = boost::make_shared<ParameterMap>(*getParameterMap());
+  pmap->setDetectorInfo(nullptr);
+  pmap->setInstrument(nullptr);
+
+  if (!hasDetectorInfo())
+    return pmap;
+
+  const auto &baseInstr = m_map ? *m_instr : *this;
+  for (size_t i = 0; i < m_detectorInfo->size(); ++i) {
+    const auto &det = std::get<1>(baseInstr.m_detectorCache[i]);
+    if (m_detectorInfo->isMasked(i)) {
+      pmap->forceUnsafeSetMasked(det.get(), true);
+    }
+
+    // Compare detector position with position based on pmap. Note that we
+    // previously stripped the DetectorInfo from pmap so we do not get positions
+    // from the DetectorInfo but only those directly stored in pmap.
+    const auto parDet = ParComponentFactory::create(det, pmap.get());
+    auto relPos = m_detectorInfo->position(i);
+    auto relRot = m_detectorInfo->rotation(i);
+    if (const auto parent = parDet->getParent()) {
+      relPos -= toVector3d(parent->getPos());
+      const auto invParentRot =
+          toQuaterniond(parent->getRotation()).conjugate();
+      relPos = invParentRot * relPos;
+      // Note the unusual order, see Component.cpp
+      relRot = invParentRot * relRot;
+    }
+    // Tolerance 1e-6 m. Is this a good value?
+    if ((relPos - toVector3d(parDet->getRelativePos())).norm() > 1e-6) {
```

Comment:
I think I will actually change it to the value used in the comparison method for `DetectorInfo` (`1e-9m`). To make it fully equivalent I might also need to change the tolerance check here to be equivalent to the more sophisticated approach in `DetectorInfo`.